src/react/databrowser/objects/__tests__/Tags.test.jsx

```diff
@@ -0,0 +1,124 @@
+import * as s3objects from '../../../actions/s3object';
+import { Item } from '../../../ui-elements/EditableKeyValue';
+import { OBJECT_METADATA } from '../../../actions/__tests__/utils/testUtil';
+import React from 'react';
+import Tags from '../details/Tags';
+import { reduxMount } from '../../../utils/test';
+
+describe('Tags', () => {
+    const putObjectTaggingMock = jest.spyOn(s3objects, 'putObjectTagging').mockReturnValue({ type: '' });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('Tags should render', async () => {
+        const { component } = reduxMount(<Tags objectMetadata={OBJECT_METADATA}/>);
+
+        expect(component.find(Tags).isEmptyRender()).toBe(false);
+    });
+
+    it('should render by default an Item with empty values in each input when there are no key/value present', () => {
+        const { component } = reduxMount(<Tags objectMetadata={OBJECT_METADATA}/>);
+
+        // check if only one item row is rendered
+        const items = component.find(Item);
+        expect(items).toHaveLength(1);
+
+        // check if item row values are rendered
+        const firstItem = items.first();
+        expect(firstItem.find('input#tags-input-key').prop('value')).toBe('');
+        expect(firstItem.find('input#tags-input-value').prop('value')).toBe('');
+    });
+
+    it('should add new key/value tag and should trigger function if save button is pressed', () => {
+        const { component } = reduxMount(<Tags objectMetadata={OBJECT_METADATA}/>);
+
+        // check if only one item row is rendered
+        expect(component.find(Item)).toHaveLength(1);
+
+        // fill input key and input value
+        component.find('input#tags-input-key').simulate('change', { target: { name: 'key', value: 'key1' } });
```

Comment:
We can use `firstItem` instead of `component` to make sure we are in the first Item although there is only one here.
Same thing for the following assertions.

Reply:
done